airflow/sensors/base.py

```diff
@@ -41,6 +42,13 @@
 # See  https://github.com/apache/airflow/issues/16035
 from airflow.utils.decorators import apply_defaults
 
+# As documented in https://dev.mysql.com/doc/refman/5.7/en/datetime.html.
+_MYSQL_TIMESTAMP_MAX = datetime.datetime(2038, 1, 19, 3, 14, 7, tzinfo=timezone.utc)
+
+
+def _is_metadatabase_mysql() -> bool:
+    return settings.engine and settings.engine.url.get_backend_name() == "mysql"
```

Comment:
Cache this? Move it in to settings.py?

Reply:
Adding cache. It doesn’t seem to worth putting in `settings.py` for me, unless we can find more use cases for the function.